src/adapter/sources.ts

```diff
@@ -23,7 +23,13 @@ export interface UiLocation {
   lineNumber: number; // 1-based
   columnNumber: number;  // 1-based
   source: Source;
-};
+}
+
+function isUiLocation(loc: any): loc is UiLocation {
```

Comment:
`unknown` would be the more correct type to use here, TS will force you to narrow it safely :)